lib/synapse/haproxy.rb

```diff
@@ -743,12 +752,10 @@ def update_backends(watchers)
           rescue StandardError => e
             log.warn "synapse: unknown error writing to socket"
             @restart_required = true
-            return
           else
             unless output == "\n"
```

Comment:
You're right.  On a related note, I'm guessing that you removed the two `return`s so that we're able to make as much progress as possible, even in the case of errors?